lib/common/avb.c

```diff
@@ -87,47 +93,104 @@ int pci_connect(device_t *igb_dev)
 
-int gptpinit(int *shm_fd, char **memory_offset_buffer)
+/**
+ * @brief Open the memory mapping used for IPC
+ * @param shm_fd [inout] File descriptor for mapping
+ * @param shm_map [inout] Pointer to mapping
+ * @return 0 for success, negative for failure
+ */
+ 
+int gptpinit(int *shm_fd, char **shm_map)
 {
+	if (NULL == shm_fd || NULL == shm_map) {
+		return -1;
+	}
 	*shm_fd = shm_open(SHM_NAME, O_RDWR, 0);
 	if (*shm_fd == -1) {
 		perror("shm_open()");
-		return false;
+		return -1;
 	}
-	*memory_offset_buffer =
-	    (char *)mmap(NULL, SHM_SIZE, PROT_READ | PROT_WRITE, MAP_SHARED,
-			 *shm_fd, 0);
-	if (*memory_offset_buffer == (char *)-1) {
+	*shm_map = (char *)mmap(NULL, SHM_SIZE, PROT_READ | PROT_WRITE,
+				MAP_SHARED, *shm_fd, 0);
+	if ((char*)-1 == *shm_map) {
 		perror("mmap()");
-		*memory_offset_buffer = NULL;
+		*shm_map = NULL;
 		shm_unlink(SHM_NAME);
-		return false;
+		return -1;
 	}
-	return true;
+	return 0;
 }
 
-void gptpdeinit(int shm_fd, char *memory_offset_buffer)
-{
-	if (memory_offset_buffer != NULL) {
-		munmap(memory_offset_buffer, SHM_SIZE);
+/**
+ * @brief Free the memory mapping used for IPC
+ * @param shm_fd [in] File descriptor for mapping
+ * @param shm_map [in] Pointer to mapping
+ * @return 0 for success, negative for failure
+ *	-1 = close failed
+ *	-2 = munmap failed
+ *	-3 = close and munmap failed
+ */
+
+int gptpdeinit(int *shm_fd, char **shm_map)
+{
+	int ret = 0;
+	if (NULL == shm_fd || -1 == *shm_fd) {
+		ret -= 1;
+	} else {
+		if(close(*shm_fd) == -1) {
+			ret -= 1;
+		}
+		*shm_fd = -1;
 	}
-	if (shm_fd != -1) {
-		close(shm_fd);
+	if (NULL == shm_map || NULL == *shm_map) {
+		ret -= 2;
+	} else {
+		if (munmap(*shm_map, SHM_SIZE) == -1) {
+			ret -= 2;
+		}
+		*shm_map = NULL;
 	}
+	return ret;
 }
 
-int gptpscaling(gPtpTimeData * td, char *memory_offset_buffer)
+/**
+ * @brief Read the ptp data from IPC memory
+ * @param shm_map [in] Pointer to mapping
+ * @param td [inout] Struct to read the data into
+ * @return 0 for success, negative for failure
+ */
+
+int gptpgetdata(char *shm_map, gPtpTimeData *td)
 {
-	pthread_mutex_lock((pthread_mutex_t *) memory_offset_buffer);
-	memcpy(td, memory_offset_buffer + sizeof(pthread_mutex_t), sizeof(*td));
-	pthread_mutex_unlock((pthread_mutex_t *) memory_offset_buffer);
+	if (NULL == shm_map || NULL == td) {
+		return -1;
+	}
+	pthread_mutex_lock((pthread_mutex_t *) shm_map);
+	memcpy(td, shm_map + sizeof(pthread_mutex_t), sizeof(*td));
+	pthread_mutex_unlock((pthread_mutex_t *) shm_map);
+
+	return 0;
+}
 
+/**
+ * @brief Read the ptp data from IPC memory and print its contents
+ * @param shm_map [in] Pointer to mapping
+ * @param td [inout] Struct to read the data into
+ * @return 0 for success, negative for failure
+ */
+
+int gptpscaling(char *shm_map, gPtpTimeData *td) //change this function name ??
+{
+	int i;
+	if ((i = gptpgetdata(shm_map, td)) < 0) {
+		return i;
+	}
 	fprintf(stderr, "ml_phoffset = %" PRId64 ", ls_phoffset = %" PRId64 "\n",
```

Comment:
I created gptpgetdata to avoid the printfs, any reason it is called gptpscaling ?